File: app/src/main/java/me/cafecode/android/newspaper/newses/NewsesContract.java
package me.cafecode.android.newspaper.newses;

import java.util.List;

import me.cafecode.android.newspaper.BasePresenter;
import me.cafecode.android.newspaper.BaseView;
import me.cafecode.android.newspaper.data.models.News;

/**
 * Created by Natthawut Hemathulin on 5/31/2017 AD.
 * Email: natthawut1991@gmail.com
 */

public interface NewsesContract {

    interface View extends BaseView {

        void showProgressView(boolean isShow);

        void showNewses(List<News> newses);

    }

    interface Presenter extends BasePresenter {

        void loadNewses();

    }

}

File: app/src/main/java/me/cafecode/android/newspaper/newses/NewsesContract.kt
package me.cafecode.android.newspaper.newses

import me.cafecode.android.newspaper.BasePresenter
import me.cafecode.android.newspaper.BaseView
import me.cafecode.android.newspaper.data.models.News

interface NewsesContract {

    interface View : BaseView<Presenter> {

        fun showProgressView(isShow: Boolean)

        fun showNewses(newses: List<News>)

        fun showErrorMessage(message: String)

    }

    interface Presenter : BasePresenter {

        fun loadNewses()

    }
}